extensions/reboot-to-bios/src/reboot-to-bios.ts

```diff
@@ -0,0 +1,59 @@
+import { showHUD, confirmAlert, getPreferenceValues } from "@raycast/api";
+import { showFailureToast } from "@raycast/utils";
+import { exec } from "child_process";
+import { promisify } from "util";
+import { writeFile } from "fs/promises";
+import { tmpdir } from "os";
+import { join } from "path";
+
+const execAsync = promisify(exec);
+
+interface Preferences {
+  showConfirmation: boolean;
+}
+
+async function executeReboot() {
+  const batPath = join(tmpdir(), "reboot_to_bios.bat");
+  const batContent = `@echo off
+if not DEFINED IS_MINIMIZED set IS_MINIMIZED=1 && start "" /min "%~f0" %* && exit
+shutdown /r /fw /t 0
+`;
+  await writeFile(batPath, batContent);
+
+  // Show success message before executing, as the system will reboot immediately
+  await showHUD("Rebooting to BIOS...");
+
+  // Execute the batch file - ignore any errors as the system will be rebooting
+  // and explorer.exe may not exit cleanly
+  try {
+    await execAsync(`explorer.exe "${batPath}"`);
+  } catch {
+    // Ignore errors - the reboot command works even if explorer.exe fails to return
+  }
+}
```

Comment:
You don't need to create a new file anymore in the temporaty directory. It was accumulate, it is better to add the new file to the extension directory and if there is one ,just receive it :)

```suggestion
async function getOrCreateBatFile(): Promise<string> {
  const batPath = join(environment.supportPath, "reboot_to_bios.bat");
  
  try {
    await access(batPath);
    return batPath;
  } catch {
    const batContent = `@echo off
if not DEFINED IS_MINIMIZED set IS_MINIMIZED=1 && start "" /min "%~f0" %* && exit
shutdown /r /fw /t 0
(goto) 2>nul & del "%~f0"
`;
    await writeFile(batPath, batContent);
    return batPath;
  }
}
async function executeReboot() {
   const batPath = await getOrCreateBatchFile();

  // Show success message before executing, as the system will reboot immediately
  await showHUD("Rebooting to BIOS...");

  // Execute the batch file - ignore any errors as the system will be rebooting
  // and explorer.exe may not exit cleanly
  try {
    await execAsync(`explorer.exe "${batPath}"`);
  } catch {
    // Ignore errors - the reboot command works even if explorer.exe fails to return
  }
}
```